### ISSUE TITLE
implementation of hinge_loss

### ISSUE THREAD

**Issue by wangyongjie-ntu** [interpretml/DiCE#288]
Can anyone help explain why the implementation of hinge_loss is different from the description in the paper? is it necessary to convert probability into logits first, then compute the difference between target class and logits?  line 244 - 252 in https://github.com/interpretml/DiCE/blob/master/dice_ml/explainer_interfaces/dice_pytorch.py.  Why not directly compute the difference between target probability and predicted probability? Is there any advantage in the current implementation?
